distage/distage-framework/src/main/scala/izumi/distage/effect/modules/PolymorphicCatsDIEffectModule.scala

```diff
@@ -0,0 +1,46 @@
+package izumi.distage.effect.modules
+
+import cats.effect.{Async, Bracket, Concurrent, ConcurrentEffect, Sync, Timer}
+import cats.{Applicative, Functor, Monad, MonadError, Parallel}
+import distage.{ModuleDef, TagK}
+import izumi.distage.model.effect.{DIApplicative, DIEffect, DIEffectAsync, DIEffectRunner}
+import izumi.functional.mono.SyncSafe
+import monix.bio.Task
+
+/**
+  * Module definition that binds effect TC instances for an arbitrary F[_].
+  *
+  */
+class PolymorphicCatsDIEffectModule[F[_]: ConcurrentEffect: TagK] extends ModuleDef {
```

Comment:
> `: ConcurrentEffect`

I think it's a bad idea to require ConcurrentEffect to be an external value already created somehow, instead of expecting it to come from DI (where it can be configured)

So, instead of requiring this instance as a value, you may require it in each binding separately, e.g.

instead of 
```scala
  addImplicit[Functor[F]]
  addImplicit[Applicative[F]]
  addImplicit[Monad[F]]
```
->
```scala
  make[Functor[F]].using[ConcurrentEffect[F]]
  make[Applicative[F]].using[ConcurrentEffect[F]]
  make[Monad[F]].using[ConcurrentEffect[F]]
```

and instead of

```scala
  addImplicit[DIEffect[F]]
```
->
```scala
  make[DIEffect[F]].from { implicit F: Sync[F] => DIEffect.fromCatsEffect }
```